packages/popover/src/vaadin-popover-target-mixin.js

```diff
@@ -82,14 +87,16 @@ export const PopoverTargetMixin = (superClass) =>
     }
 
     /** @private */
-    __targetChanged(target, oldTarget) {
-      if (oldTarget) {
-        this._removeTargetListeners(oldTarget);
+    __targetOrConnectedChanged(target, isConnected) {
+      if (this.__previousTarget && (this.__previousTarget !== target || !isConnected)) {
```

Comment:
The `this.__previousTarget !== target` condition isn't tested.

Maybe:
```js
it('should not open if target is cleared', async () => {
  popover.target = target;
  await nextUpdate(popover);
  popover.target = null;
  await nextUpdate(popover);

  target.click();

  expect(popover.opened).to.be.false;
});
```

Reply:
Thanks, added a test.